src/librustc/mir/interpret/error.rs

```diff
@@ -460,13 +460,9 @@ impl<'tcx, O: fmt::Debug> fmt::Debug for InterpError<'tcx, O> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         use self::InterpError::*;
         match *self {
-            PointerOutOfBounds { ptr, check, allocation_size } => {
+            PointerOutOfBounds { ptr, msg, allocation_size } => {
                 write!(f, "Pointer must be in-bounds{} at offset {}, but is outside bounds of \
```

Comment:
This will not format very well, will it? "Pointer must be in-boundsmemory access at offset ...".

Please try and make it so that the result is a proper sentence for all possible values of `msg`.

Reply:
now it is displayed like "Inbounds test failed: ...", "Null Pointer test failed: ..."

Reply:
"Memory access test failed" looks weird though. Maybe move the "test" part to `CheckInAllocMsg::fmt`, and make the general format string just "{} test failed: pointer must be..."?